examples/Protonect.cpp

```diff
@@ -48,6 +48,27 @@ void sigint_handler(int s)
   protonect_shutdown = true;
 }
 
+bool protonect_paused = false;
+libfreenect2::Freenect2Device *devtopause;
+
+//Doing non-trivial things in signal handler is bad. If you want to pause,
+//do it in another thread.
+//Though libusb operations are generally thread safe, I cannot guarantee
+//everything above is thread safe when calling start()/stop() while
+//waitForNewFrame().
+void sigusr1_handler(int s)
+{
+  if (devtopause == 0)
+    return;
+/// [pause]
+  if (protonect_paused)
```

Comment:
I don't have time to implement it at the moment just for an example.

I think a good solution is to use waitForNewFrame(frame, milliseconds), which needs `wait_for()`, a C++11 feature. This would need backporting `wait_for()` to tinythread. I have no idea how hard that it.